cmake/defaults/Packages.cmake

```diff
@@ -211,8 +211,16 @@ if (PXR_BUILD_IMAGING)
         find_package(OpenImageIO REQUIRED)
         set(OIIO_VERSION "${OpenImageIO_VERSION_MAJOR}.${OpenImageIO_VERSION_MINOR}.${OpenImageIO_VERSION_PATCH}")
         add_definitions(-DPXR_OIIO_PLUGIN_ENABLED)
+        if (CMAKE_BUILD_TYPE)
+            string(TOUPPER "${CMAKE_BUILD_TYPE}" UPPER_BUILD_TYPE)
```

Comment:
This is a global setting, so wouldn't it make sense to do it more globally? I don't know if anything else references the CMAKE_BUILD_TYPE, but it might be weird for it to conditionally switch in the middle of the build configuration.

Reply:
I'm looking at issues at cmake ~ e.g. https://gitlab.kitware.com/cmake/cmake/-/issues/22591 ~ and it leaves me a bit confused. It seems like things are case sensitive, but should match e.g. Release or Debug, etc. Is the problem that OIIO is expecting RELEASE and DEBUG, etc? If so, perhaps there's an issue to file with OIIO?

Reply:
We're not changing `CMAKE_BUILD_TYPE` but use its value with upper case to find the proper `IMPORTED_LOCATION_XXX ` property of OIIO.

[cmake-build-type](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html#cmake-build-type) says that its typical values include Debug, Release, RelWithDebInfo and MinSizeRel, but custom build types can also be defined, and maybe treated case-sensitively or case-insensitively depending on the situation. But property IMPORTED_LOCATION_XXX is always in upper case, so we need to use upper case to find the property defined when creating OIIO targets.